ansible/roles/etesync/tasks/main.yml

```diff
@@ -0,0 +1,204 @@
+  tags: [ 'role::etesync:config' ]
+
+- name: Generate extended EteSync configuration
+  template:
+    src: 'usr/local/lib/etesync/app/etesync_site_settings.py.j2'
+    dest: '{{ etesync__git_checkout + "/etesync_site_settings.py" }}'
+    owner: '{{ etesync__user }}'
+    group: '{{ etesync__group }}'
+    mode: '0640'
+  register: etesync__register_config
+  notify: [ 'Restart gunicorn for etesync' ]
+  tags: [ 'role::etesync:config' ]
+
+- name: Generate EteSync secret.txt file
+  template:
+    src: 'etc/etesync-server/secret.txt.j2'
+    dest: '{{ etesync__config_secret_key_filepath }}'
+    owner: 'root'
+    group: '{{ etesync__group }}'
+    mode: '0640'
+  notify: [ 'Restart gunicorn for etesync' ]
+  tags: [ 'role::etesync:config' ]
+
+- name: Perform database installation or migration
+  shell: |
+    ## -o nounset can be enabled with virtualenv >= 16.2.
+    ## Ref: https://github.com/pypa/virtualenv/pull/922
+    set -o pipefail -o errexit;
+    source {{ etesync__virtualenv }}/bin/activate
+    ./manage.py migrate
+  args:
+    chdir: '{{ etesync__git_checkout }}'
+    executable: 'bash'
+  become: True
+  become_user: '{{ etesync__user }}'
+  changed_when: ("No migrations to apply." not in etesync__register_migration.stdout)
+  when: etesync__register_source is changed or etesync__register_config is changed
+  register: etesync__register_migration
+
+- name: Create superuser account
+  environment:
+    SUPERUSER_NAME: '{{ etesync__superuser_name }}'
+    SUPERUSER_EMAIL: '{{ etesync__superuser_email }}'
+    SUPERUSER_PASSWORD: '{{ etesync__superuser_password }}'
+  shell: |
+    ## -o nounset can be enabled with virtualenv >= 16.2.
+    ## Ref: https://github.com/pypa/virtualenv/pull/922
+    set -o pipefail -o errexit;
+    source {{ etesync__virtualenv }}/bin/activate
+    echo "from django.contrib.auth.models import User;
+          User.objects.create_superuser('${SUPERUSER_NAME}', '${SUPERUSER_EMAIL}', '${SUPERUSER_PASSWORD}')" | ./manage.py shell
+  args:
+    chdir: '{{ etesync__git_checkout }}'
+    executable: 'bash'
+  become: True
+  become_user: '{{ etesync__user }}'
+  when: (not etesync__register_installed.stat.exists|bool and
+         etesync__register_migration is not changed)
+  no_log: True
+
+- name: Make sure that Ansible local facts directory exists
+  file:
+    path: '/etc/ansible/facts.d'
+    state: 'directory'
+    owner: 'root'
+    group: 'root'
+    mode: '0755'
+
+- name: Save EteSync local facts
+  template:
+    src: 'etc/ansible/facts.d/etesync.fact.j2'
+    dest: '/etc/ansible/facts.d/etesync.fact'
+    owner: 'root'
+    group: 'root'
+    mode: '0755'
+  register: etesync__register_facts
+
+- name: Update Ansible facts if they were modified
+  action: setup
+  when: etesync__register_facts is changed
+
+- name: Print URL entry point if EteSync is deployed on a subpath
+  debug:
+    msg: 'URL: {{ etesync__url }}'
+  when: etesync__http_psk_subpath != ""
+  tags: [ 'role::etesync:show_url' ]
```

Comment:
I don't like to have `debug` modules in the production code, this implies that Ansible output is meant to be read and interactive which is not strictly true. We need to find a better, general purpose solution for this. Perhaps a mail to `root` on first execution would suffice? This would also ensure that the message with the details can be archived if necessary. The SMTP configuration should make sure that it will be sent to the correct person.

Reply:
Very good idea, thanks. I don’t operate (with DebOps/Ansible) at such a scale but I can definitely see this come in handy. Done.